frontend/src/components/common/protected-route/protected-route.tsx

```diff
@@ -13,13 +14,17 @@ const ProtectedRoute: FC<Props> = ({
   redirectTo = AppRoute.SIGN_IN,
   component,
 }) => {
-  const { user } = useAppSelector(({ auth }) => ({
-    user: auth.user,
-  }));
+  const { user, dataStatus } = useAppSelector((state) => state.auth);
 
   const hasUser = Boolean(user);
+  const token = Boolean(storage.getItem(StorageKey.TOKEN));
 
-  if (!hasUser) {
+  if (
+    !token ||
+    (!hasUser &&
+      (dataStatus === DataStatus.FULFILLED ||
+        dataStatus === DataStatus.REJECTED))
+  ) {
```

Comment:
You can store this check within `hasUser` variable and move dataStatus checks to separate variable to make it more readable